src/core/build_simulations.jl

```diff
@@ -80,17 +80,29 @@ function _build_stages(sim_ref::SimulationRef,
 
     for (k, v) in stages
         verbose && @info("Building Stage $(k)")
-        op_mod = OperationModel(DefaultOpModel, v.model, v.sys;
-                                optimizer = v.optimizer,
-                                parameters = true,
-                                verbose = verbose,
-                                kwargs...)
+        canonical = _build_canonical(v.model.transmission,
+                                    v.model.devices,
+                                    v.model.branches,
+                                    v.model.services,
+                                    v.sys,
+                                    v.optimizer,
+                                    verbose;
+                                    parameters = true,
+                                    kwargs...)
         stage_path = joinpath(sim_ref.models,"stage_$(k)_model")
         mkpath(stage_path)
-        write_op_model(op_mod, joinpath(stage_path, "optimization_model.json"))
+        write_op_model(canonical, joinpath(stage_path, "optimization_model.json"))
         system_to_file && IS.to_json(v.sys, joinpath(stage_path ,"sys_data.json"))
-        _populate_cache!(v.cache, op_mod)
-        mod_stages[k] = _Stage(k, op_mod, v.execution_count, v.chronology_ref, v.cache)
+        _populate_cache!(v.cache, canonical)
+        mod_stages[k] = _Stage(k,
```

Comment:
It looks like almost everything you need to construct _Stage is in a Stage. Could you just pass "v" to this constructor instead of all of these members?

Reply:
Yes, the idea is to make only one. For now, I just needed more fields in _Stage to implement the capability of re-building a stage in each run.